src/components/OutputCard.vue

```diff
@@ -0,0 +1,117 @@
+<script setup>
+import { computed } from 'vue'
+
+import JSZip from 'jszip'
+import { saveAs } from 'file-saver'
+
+import CodeView from '@/components/CodeView.vue'
+
+const props = defineProps({
+  info: {
+    type: Object,
+    required: true
+  },
+  keymap: {
+    type: String,
+    required: true
+  },
+  rules: {
+    type: String,
+    required: true
+  }
+})
+
+const prettyInfo = computed(() => {
+  if (props.info) {
+    return (
+      JSON.stringify(props.info, null, '    ')
+        // Place layout keys on a single line
+        .replace(/\n {20,}/g, ' ')
+        .replace(/\n {16,}\}/g, ' }')
+        // Remove spacing
+        .replace(/\{ "label"/g, '{"label"')
+        .replace(/([0-9.]+) \}/g, '$1}')
+        // Place matrix pin arrays on a single line
+        .replace(/\n {12}"([BCDEF][0-7])"/g, ' "$1"')
+        .replace(/"\n +\]/g, '" ]')
+        // Remove spacing
+        .replace(/\[ /g, '[')
+        .replace(/ \]/g, ']') + '\n'
+    )
+  }
+
+  return ''
+})
+
+function downloadZip() {
+  const zip = new JSZip()
+
+  zip.file('info.json', prettyInfo.value)
+  zip.file('rules.mk', props.rules)
+  zip.folder('keymaps').folder('default').file('keymap.c', props.keymap)
+
+  zip
+    .generateAsync({
+      type: 'blob'
+    })
+    .then((content) => {
+      saveAs(content, 'keyboard.zip')
```

Comment:
Would it be possible to name the ZIP file to match the name of the JSON file that was imported, such that importing e.g. `endless80.json` results in an `endless80.zip` at the end?

Reply:
Good point.